Segment store server process should exit upon unrecoverable startup failure

**Problem description**
I observe that startup errors encountered in the SS `ServiceStarter` leave the process running but unusable.    It should exit so that it recovers.

For example, if the segment store is started before BookKeeper is fully initialized, the ZK ledger path might be missing, which causes the below exception and leaves the SSS running but non-functional.

```
2018-03-07 18:26:56,386 1593 [main] INFO  i.p.s.server.host.ServiceStarter - ZooKeeper Client shut down.
Exception in thread "main" io.pravega.segmentstore.storage.DataLogNotAvailableException: Unable to establish connection to ZooKeeper or BookKeeper.
        at io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory.initialize(BookKeeperLogFactory.java:96)
        at io.pravega.segmentstore.server.store.ServiceBuilder.initialize(ServiceBuilder.java:217)
        at io.pravega.segmentstore.server.host.ServiceStarter.start(ServiceStarter.java:94)
        at io.pravega.segmentstore.server.host.ServiceStarter.main(ServiceStarter.java:227)
Caused by: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /pravega/pravega-cluster/bookkeeper/ledgers/available/readonly
        at org.apache.zookeeper.KeeperException.create(KeeperException.java:115)
        at org.apache.zookeeper.KeeperException.create(KeeperException.java:51)
        at org.apache.zookeeper.ZooKeeper.create(ZooKeeper.java:1426)
        at org.apache.bookkeeper.zookeeper.ZooKeeperClient.access$1801(ZooKeeperClient.java:71)
        at org.apache.bookkeeper.zookeeper.ZooKeeperClient$9.call(ZooKeeperClient.java:695)
        at org.apache.bookkeeper.zookeeper.ZooKeeperClient$9.call(ZooKeeperClient.java:689)
        at org.apache.bookkeeper.zookeeper.ZooWorker.syncCallWithRetries(ZooWorker.java:141)
        at org.apache.bookkeeper.zookeeper.ZooKeeperClient.create(ZooKeeperClient.java:689)
        at org.apache.bookkeeper.client.BookieWatcher$ReadOnlyBookieWatcher.<init>(BookieWatcher.java:345)
        at org.apache.bookkeeper.client.BookieWatcher.<init>(BookieWatcher.java:98)
        at org.apache.bookkeeper.client.BookKeeper.<init>(BookKeeper.java:486)
        at org.apache.bookkeeper.client.BookKeeper.<init>(BookKeeper.java:321)
        at io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory.startBookKeeperClient(BookKeeperLogFactory.java:132)
        at io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory.initialize(BookKeeperLogFactory.java:84)
        ... 3 more
```

--- a/segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/ServiceStarter.java
+++ b/segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/ServiceStarter.java
@@ -227,6 +227,12 @@ public final class ServiceStarter {
 
         try {
             serviceStarter.get().start();
+        } catch (Throwable e) {
+            log.error("Could not start the Service, Aborting.", e);
+            System.exit(1);
+        }
+
+        try {
             Runtime.getRuntime().addShutdownHook(new Thread(() -> {
                 log.info("Caught interrupt signal...");
                 serviceStarter.get().shutdown();
@@ -237,6 +243,7 @@ public final class ServiceStarter {
             log.info("Caught interrupt signal...");
         } finally {
             serviceStarter.get().shutdown();
+            System.exit(0);
         }
     }
 

